tests/ipex/test_modeling.py

```diff
@@ -255,6 +259,37 @@ def test_pipeline(self, model_arch):
         self.assertEqual(pipe.device, model.device)
         self.assertTrue(all("This is a sample" in item["generated_text"] for item in outputs))
 
+    @parameterized.expand(
+        grid_parameters(
+            {
+                "model_arch": IPEX_PATCHED_SUPPORTED_ARCHITECTURES,
+                "use_cache": [True, False],
+                "num_beams": [1, 4],
+                "batch_size": [1, 4],
+            }
+        )
+    )
+    @unittest.skipIf(is_ipex_version("<=", "2.3.0"), reason="Only ipex version > 2.3.0 supports ipex model patching")
+    def test_ipex_patching(self, test_name, model_arch, use_cache, num_beams, batch_size):
+        model_id = MODEL_NAMES[model_arch]
+        set_seed(SEED)
+        model = IPEXModelForCausalLM.from_pretrained(model_id, export=True)
```

Comment:
the model will always be exported with `use_cache=True` we should instead set `use_cache` :
```suggestion
        model = IPEXModelForCausalLM.from_pretrained(model_id, export=True, use_cache=use_cache)
```

can you also add a check 

```suggestion
        model = IPEXModelForCausalLM.from_pretrained(model_id, export=True, use_cache=use_cache)
        self.assertTrue(model.use_cache, use_cache)
```